### DISCUSSION TITLE
Baldur's Gate Style Item Inspection / Statistic Window possible; yes (book window)?

### DISCUSSION THREAD

**Discussion by WithinAmnesia** [flareteam/flare-game#911] [Q&A] [answered]
Can the Flare engine do something like this? https://youtu.be/pUXLRkH2mps?list=PLnP7DpDaY9fvBEjTUUtY4yQjTIQZpJoHS&t=1517
An inspect / right clink / 'open / read ~book' option that has a picture and scroll-able text (as well as the ~'mouse hover-over item stats tool tip' found in Flare / Diablo / WoW)?

I think this 'item inspect / read item' feature can be done already in Flare. Although I hope to eventually have every item in my game with a ~'right click to inspect / read item' option. As in like how every item in Baldur's Gate has an 'inspect / read' right click window with an item description and an item description picture format.

I am hoping that I can have the tool-tip stats display of Diablo / WoW / Flare with the option to 'inspect / read item' like Baldur's Gate;  hopefully the best of both worlds. Although I am unsure if that is technically reasonable for every item with potentially 10,000+ items hosted within the Flare engine? Can this be done or am I missing something?

I am pretty sure I am ~90-95% ~9/10+ 'done' with gameplay feature / stress testing / engine capability testing. I am trying to hopefully avoid the dreaded production hell ~'80 / 20 ratio' of where the last 20% of the project's work has 80% of the projects difficulty / work x.X. So with that said I hope that one can understand why I am trying very hard to hopefully avoid feature bloat X-P. 

So hopefully before I go crazy with the art asset production, (for 'finished' art production cycles can be near infinite it seems at times) I should be reasonable in my work flow pipeline / workload / mental health; and hopefully standardize core gameplay asset mass production (items, creatures and abilities). So as to hopefully keep my monotonous work input low and creative production out put high. I better work efficiently or else I am in production output trouble, for I am just one guy lol.

I am trying to be responsible and develop my core gameplay systems first then the art. So as to hopefully stay on track this time (not like my ~half decade of failed RPG Maker Projects x.X). So as to actually release something that hopefully works well with high quality gameplay experiences and that realistically 'gets across the line'; so that people can hopefully enjoy the game verses (for what seems / feels like) forever being in development. I started what eventually morphed into this incarnation of this RPG project in at least the middle 2000's. Or what the hell, it was the late 90's if I am really being honest but it has sure grown and evolved since then X-P. Thus I am waiting on the reworked stats mechanic systems to come to due time (they come when they come; no rush, quality > quantity).

Although this 'read item / 'inspect item' feature I would like to make standard on all inventory items in my game (with the ~'video games gods' / bug test rounds willing). Yet I think the in-depth item description / 'inspect item' feature can be technically reasonable (as in to make all of my items operate like 'Flare books' with also stat tool tips like Diablo / WoW). Yet I am not 100% confident in that assessment thus I am asking for feedback here :-).

Anyway thank you for reading all of this wordy / lengthy text. Any suggestions / feedback is very much welcome. The more perspectives the better; usually anyway X-D.

So do you think that a Baldur's  Gate style item inspection window can work en mass for the Flare engine? Can the ~'Flare book system' handle all of this? Can the Diablo / WoW / Flare ~'mouse hover-over item stats tool tip' also work along side / in conjunction with this proposed ~'right click inspect / read item' system? @dorkster

**Comment by dorkster** [comment, marked as answer]
I think you can accomplish something close to what you want by adding a `book=...` property to each item. There are a few caveats:

- The tooltip text says "Press [Right Mouse] to read", which is a bit inaccurate in this context.
- You can't "read" the item when it's equipped. I think this is just an oversight with the inventory menu code and should be fixable.
- There's currently no way to create a scrolling areas within books. This by itself is desirable functionality in my opinion.

**Reply by WithinAmnesia** [reply]
Okay that sounds encouraging, so this feature is reasonable to tinker with for it can probably work with a bit of tweaking? 

So is the scrolling in books related to menu scrolling or is it's own thing? I think we have pages (that could work?) Although admittedly ~'book scrolling' would be nice.

Yet I don't want to keep asking for ~'_A B C, X Y Z_' for I hope to stay reasonable in scope / avoid feature bloat / creep. Yet I suppose if nothing drastic / crazy / way too overly complex is involved (like a little tweak) to get ~'book scrolling' to work; then I guess that could still be reasonable?

Also is getting equipped 'books' / equipped readable items working also a sort of ~'tweak' to get working or is it one of those wild goose chase / spaghetti code / overly way too overly complex kind of things? I am just not sure how complex these minor features are to get working or not.

Although I suppose asking / looking for feedback it is just a conversation. So hopefully is not too much trouble X-P. So I think / hope that the book / read / inspect item system can work / is worth pursuing in the near future. Item in-depth descriptions would be pretty cool :-P.

I would like to have the book / read / inspect item system in my game but it is more of a third line / tertiary priority for myself I feel. Like it would be nice to have but if it does work out for 1 2 3 or A B C reason(s) then it is not the end of the world. Yet I really those in depth story / lore / item description for items, little bits of optional lore just seems to make the world a b it more more alive.

What do you think about this kind of Baldur's Gate read / inspect item system? Do you think people will like it? I hope other people can enjoy it, like extra lore / details / neat like quality touches to hopefully make game worlds feel more alive / give them a soul?

I wonder if people could make like a detective / slower paced content with this like where item descriptions are important type thing? Like I know Flare is like Diablo II / action combat paced, yet I wonder what else could be done. Like that Farming project, I wonder if in-depth item descriptions could help those kinds of projects as well?

I don't know all the details (it could be too hard to even make reasonably happen)? Either way implemented or not I think in-depth read / inspect item system would be cool as an option for Flare. I hope that other people do too / could find cool ways to use it X-P.

**Reply by dorkster** [reply]
>So is the scrolling in books related to menu scrolling or is it's own thing? I think we have pages (that could work?) Although admittedly ~'book scrolling' would be nice.

Kind of its own thing. I think it would be useful for books that behave as menus (e.g. flare-game's Alchemy menu), or for things that aren't entirely physical (e.g. a "PDA data log" in a sci-fi game)

>What do you think about this kind of Baldur's Gate read / inspect item system? Do you think people will like it?

I personally think it's a bit too much for an ARPG. I would first try using the putting the description in the `flavor` field of the item. If the descriptions are lengthy, you could try increasing the maximum tooltip width via the `tooltip_width` property in `engine/tooltips.txt`.

**Reply by WithinAmnesia** [reply]
Simplicity is the ultimate form of sophistication, quantity has a quality all of its own; a true master can make something complex become simple yet retain its soul.

**Reply by WithinAmnesia** [reply]
@dorkster I think it would be better anyway to separate items and lore books for I'm going for more or less standardized quality / ~'tiers' of quality / item types / semi-procedurally generated / sticking to a standardized formula to generate / populate the items in my game.

In Baldur's Gate they had poor graphical detail with item icons being limited to 32x32 pixels for icons and 64x64 pixels when 'picking up / moving an item'. Also that game is based on AD&D where there literally are no graphics other than text description or if you are lucky a source book illustration. I think in Baldur's Gate the item inspection was a way to computerize the AD&D source material / books while also 'getting away with' bad / low fidelity graphics back when people were rocking CRT / big boxy monitors and 1080p quality graphics back then was like what 8K quality graphics is for us today XD.

So I think also with 4K standard graphics that the items don't need an illustration (which would probably just be a drawn / pencil illustration filter computer generated image up close from the base 3D model anyway). In Skyrim (I think Oblivion too? 007 Golden Eye too) the items were just displayed 'up close' / they have used the base 3D model with enough high graphical fidelity that the player can clearly see what the item looks like from the inventory and kind of obsoletes the need for item inspection windows.

Also it is kind of weird if somehow an item just some how has a book basically / effective attached to it. Also like Doom I believe that game play should come first (be a great A.R.P.G. first) and the history / lore can be optional but there if people want to read it / like that kind of stuff. Also it would give a reason to have lore book items in the game.

Also having distinct 'lore books' can have a more immersive feeling of (just how Flare is already set up) literally turning pages in a book that looks like a real book but is found in game. Also items like sword and armour should have lore books with this logic but they could be collected / found in places like a library / a blacksmith or a person trying to figure out how to craft them / places that in real life would make sense and in that way separating lore description and item stats can make a little microcosm of in world story telling and also a sort of natural optional side quest of collecting books if the player so desires.

Also this frees up a shit load of writing for every item (if there are like ~10,000 items, that is a lot of writing if every item needs an item description). Also books can be more 'meatier' where they can even go into like cultural and unreliable narrator / writer and have flavour and  a general sense of a lived in world / 'life' all to their own. Also the one thing I liked in Baldur's Gate item descriptions was the possible storied / individual item lore / in game world history attached to each unique magical item.

Yet I realize that if I have standardized item qualities / ways of improving items / almost like a repeatable science / standardize 'item tiers' then I could just make a history / lore book of how those ~'enchantments' / ways of crafting were discovered / how to craft them etc. That could be even more interesting than having a 'book' attached to an item. Like the player could find a lore book on certain types of items first and then try to 'hunt them down' in the game world. Bingo-bango natural world building, story telling and player interest and player driven and generated individual in game progression goals.

I find it kind of dumb having cursed items in the game for who are you tricking? I think identify item is kind of not needed either for like 95% of fun of getting an item I find is getting it verses like the 5% of 'Ooo what could it be?' Also Wiki's and game databases exist, so people can just look what ever / all of the items, creatures, areas, drops and drop tables up any time anyway lol. Also less can be more if what a person has is high quality and works well verses a billion crappy things that don't work XD.

I'm kind of on the fence with maps in books (formatted like Flare books). As in if the maps were made by in game world cartographers; like board scale hand written details, or even finding a secret treasure map. They would make sense but I kind of like also the Classic Everquest / M.U.D. style of game world exploration of having no map and forcing the player to use their own navigation skills. I don't like real time satellite imagery style maps in what is supposed to be a medieval fantasy style game XD.

I understand quality of life but just like having waaay too many teleportation 'ease of life' mechanics / in game realities it can cheapen / 'shrink' / take all of the fun of exploring / trying to solve your own problems with your own resources. I want to give players problems they can solve without giving them systems that do it for themselves / hand hold too too much. Some things are nice but I personally like to a form a player transformative / memorable experience via adversity and character progression in my game for people to discover / experience themselves.

So I think in the end that not having in-depth 'book like' item descriptions built into items can actually be a good thing if lore books can fill that niche and not get lazy with the writing. It can even be better than in-depth item descriptions if the proper amount of care, love, sacrifice and effort / due diligence is invested into the game world for the player to explore / discover themselves. I had some time to think this through at work today and I think having the Flare books just being 'books' is better than shoving all of the items into item-book hybrids.

Do Flare books have no limit (within reason ~10,000 pages 'max') to the amount of pages they can contain?

**Reply by dorkster** [reply]
Not all items are going to have interesting lore behind them. In these cases, it would be wiser to have a lore book that covers items as a group. For example, a book that focuses on only bladed weapons, or a book that details the clothing/armor trends in a specific region. And on the other end of the spectrum, legendary items could have books written (possibly speculative/inaccurate) about their history.

But then there's the middle area where items that aren't legendary *or* mundane exist. There'd likely be no books written about what makes these special over regular items. But I think it *would* be likely for a book to focus on a property that these items could have. For example, if magic is a thing in your world, there could be a book about magic's potential applications on weapons and armor.

----

>Do Flare books have no limit (within reason ~10,000 pages 'max') to the amount of pages they can contain?

A book in Flare is actually only a single "page". But, you can use buttons to change from one book to another. This effectively means you can have infinite pages. Although, making a 10,000 page book would be tedious for you to make and for the player to navigate. So focus on many smaller books versus one big book.

**Reply by WithinAmnesia** [reply]
@dorkster I totally agree with your points, less is more and some items are just not *that* interesting for the vast majority of people. Also another thing in favour separate books is that they can be expanded on over time freely. Verses being stuck to give every item a description; even the really simple / mundane ones.

I think realistically the max pages I would most likely end up using is like 10 with most things probably being just on a single page. Or maybe the odd book with 2 or 3 pages with probably some rare outliers with 5-6 or more pages. It is nice to have peace of mind that a book can have as many pages as desired. Yet most likely in practice 70-80%+ of 'books' will be single pages then a bell curve effect of more rare lengthy books just by the nature of writing. I like to write with lots of words (some times too many lol). Yet not everything needs tons of points of detail. It could make a lot of practical sense in having the style of some books being written in such a way that covers many areas; like you said, like bladed weapons, cultural points of interest in regards to X Y Z. Or even something like an in-world compendium / treatise on A B C and or something akin to these medieval / renaissance treaties / manuscripts https://wiktenauer.com/ / what this guy talks about https://www.youtube.com/watch?v=cJhVQpcJVss&ab_channel=scholagladiatoria . Something in that broad in-world style of writing for books would make a lot of sense and could spare game development from writing for the sake of writing. Ernest Hemingway's Old Man and the Sea took ~16 years to write and was rewritten over and over to find its literary brilliance and yet it is only 127 pages long; yet it is considered to be a masterpiece of the era https://greatbooksguy.com/2020/11/22/the-writers-endurance-the-old-man-and-the-sea/ + https://en.wikipedia.org/wiki/The_Old_Man_and_the_Sea#Background_and_publication .

I am one person, to start at least; and for likely a very long time into the foreseeable future. So it would be kind of silly / unwise if I took tons of time invested into writing books / tertiary 'fluff / juice / niceties'. Verses focusing on creating the core content that drives A.R.P.G. gameplay in with building out the main game's content. Such as spending time developing the needed map areas, abilities, creatures, art assets, animations, items, drop tables, dialog, quests, progression and very important bug fixing. So yeah I suppose that in retrospective items should be items and books should be books; specialization of content, more efficient workload, better overall content. One can hope at least; in that if one does not forget the all mighty bug fixing and polish and peer review. Also too much complexity can end up being a bad thing you know? Like having too many favours in your mouth at the same time; like pickles and ice-cream don't mix type of thing, just have pickles and then ice cream if one must devour both in the same meal X-P.
